Detectors/TOF/workflow/src/TOFClusterizerSpec.cxx

```diff
@@ -75,19 +80,35 @@ class TOFDPLClustererTask
     mClusterer.setDeltaTforClustering(mTimeWin);
 
     // initialize collision context
-    auto mcReader = std::make_unique<o2::steer::MCKinematicsReader>("collisioncontext.root");
-    auto context = mcReader->getDigitizationContext();
-    if (context) {
-      auto bcf = context->getBunchFilling();
-      std::bitset<3564> isInBC = bcf.getBCPattern();
-      for (unsigned int i = 0; i < isInBC.size(); i++) {
-        if (isInBC.test(i)) {
-          o2::tof::Utils::addInteractionBC(i, true);
+    if (gSystem->AccessPathName("collisioncontext.root")) {
+      LOG(info) << "collisioncontext.root not available, let's skip it (cosmics?) ";
+    } else {
+      auto mcReader = std::make_unique<o2::steer::MCKinematicsReader>("collisioncontext.root");
+      auto context = mcReader->getDigitizationContext();
+      if (context) {
+        auto bcf = context->getBunchFilling();
+        std::bitset<3564> isInBC = bcf.getBCPattern();
+        for (unsigned int i = 0; i < isInBC.size(); i++) {
+          if (isInBC.test(i)) {
+            o2::tof::Utils::addInteractionBC(i, true);
+          }
         }
       }
     }
   }
 
+  void finaliseCCDB(o2::framework::ConcreteDataMatcher matcher, void* obj)
+  {
+    if (matcher == ConcreteDataMatcher("TOF", "DiagnosticCal", 0)) {
+      mUpdateCCDB = true;
+    }
+    if (matcher == ConcreteDataMatcher("TOF", "LHCphaseCal", 0)) {
+      mUpdateCCDB = true;
+    }
+    if (matcher == ConcreteDataMatcher("TOF", "ChannelCalibCal", 0)) {
+      mUpdateCCDB = true;
+    }
```

Comment:
the finaliseCCDB will be called for every updated object separately: since only 1 `if` will be true, you can use `else if` or `return` from every `if` block, to avoid unnecessary comparisons.

Reply:
done